src/include/sof/audio/codec_adapter/interfaces.h

```diff
@@ -49,6 +54,18 @@ static struct codec_interface interfaces[] = {
 		.free = dummy_codec_free
 	},
 #endif /* CONFIG_DUMMY_CODEC */
+
+#if CONFIG_WAVES_CODEC
+	{
+		.id = WAVES_ID,
+		.init  = waves_codec_init,
+		.prepare = waves_codec_prepare,
+		.process = waves_codec_process,
+		.apply_config = waves_codec_apply_config,
+		.reset = waves_codec_reset,
+		.free = waves_codec_free
+	},
+#endif /* CONFIG_WAVES_CODEC */
```

Comment:
Can we not use a descriptor populated elsewhere? 

I don't think it makes any sense to have this list here. All these routines could be static and local to a wave file, just export waves_codec_interface_desc and reference it in a list, or provide a means to register it.

@lgirdwood We really need to think scalability, and this is anything but that. we want to facilitate the integration of 3rd party IP, but not have any reference to this code in the SOF tree.

TSC topic I guess...

Reply:
@plbossart yep, i already requested this to be a topic :)

Reply:
ack - I think we need to have all the codecs ops be registered (either via call or linker section). Lets include this in the TSC discussion for codecs.
However, this can be merged and fixed up after TSC discussion on the ops.

Reply:
Added to topics